### DISCUSSION TITLE
Strategy to Configure Proprietary KEM / DS algorithms

### DISCUSSION THREAD

**Discussion by quantropi-RYANTOTH** [open-quantum-safe/liboqs#1207] [Q&A]
Hello,

Thanks for the contributions to this repo -- the wiki + documentation have been really useful.

I'm struggling to determine the best way to configure a proprietary KEM / DS algorithm into this repo (locally) for subsequent testing with OPS Openssl TLS1.3 demo. My algorithms are structured the same as the other KEM/DS PQCs submitted to NIST (e.g. crypto_kem_keypair, crypto_kem_enc,crypto_kem_dec). 

I see that there are several upstream and downstream scripting tools. Is there a script I can use to automatically configure my implementations into a liboqs compatible format (e.g. generates correct function headers + updates Makefiles), or will I have to manually create and edit .c, .h, and Makefiles by referencing other PQCs?

Please let me know if you have any suggestions. Thanks!

**Comment by baentsch** [comment]
The upstream algorithm inclusion mechanism documented [here](https://github.com/open-quantum-safe/liboqs/wiki/Contributing-guide#automated-inclusion) arguably is the most automated way. If you can provide your algorithm(s) in a fitting way, most notably with YML import files like for example, [pycrystals](https://github.com/pq-crystals/kyber), adding an entry to https://github.com/open-quantum-safe/liboqs/blob/main/scripts/copy_from_upstream/copy_from_upstream.yml and running `python3 copy_from_upstream.py` should get you there... Out of curiosity: Is your algorithm proprietary or OSS?

**Reply by quantropi-RYANTOTH** [reply]
Thanks for the quick response -- the references are super useful!

Currently, our solutions are not mature enough to be OSS at this moment, but hopefully in the future we will get to that point.

**Reply by baentsch** [reply]
Looking forward to that. Can we close this discussion for the time being, then?
